### PR TITLE
[TIMOB-24664] Update to Android support library 25.3.1

### PR DESCRIPTION
**Pull request by garymathews**
- Update Android support libraries to `25.3.1`
- Set build target API from `23` -> `25`

[JIRA Ticket](https://jira.appcelerator.org/browse/TIMOB-24664)

### CODE REVIEW

File: android/modules/cardview/.classpath [hunk 1]
```diff
@@ -5,6 +5,6 @@
 	<classpathentry exported="true" kind="con" path="com.android.ide.eclipse.adt.DEPENDENCIES"/>
 	<classpathentry kind="src" path="src"/>
 	<classpathentry kind="src" path="gen"/>
-	<classpathentry exported="true" kind="lib" path="lib/cardview-v7-23.4.0.jar"/>
+	<classpathentry exported="true" kind="lib" path="lib/cardview-v7-23.0.1.jar"/>
```
Comment (hansemannn) [inline android/modules/cardview/.classpath]: From 23.4.0 to 23.0.1?

Comment (garymathews) [inline android/modules/cardview/.classpath]: Yeah, that's from reverting https://github.com/appcelerator/titanium_mobile/pull/9027/commits/80b4302cbb84690591b32837f5dea9825ac4d4f1, `.classpath` aren't used anymore. They are left-overs from Eclipse.

Also, Jenkins is failing due to Android SDK 25 not being installed.

**Review by janvennemann — CHANGES_REQUESTED** [review]

**Comment by garymathews** [comment]
@janvennemann Updated `.classpath` dependencies

**Review by janvennemann — APPROVED** [review]
CR: Pass

Approved!

**Comment by garymathews** [comment]
@janvennemann I updated this PR to add the `android.support.design` library

**Review by janvennemann — APPROVED** [review]
Looks good! 👍

**Review by janvennemann — APPROVED** [review]

**Comment by lokeshchdhry** [comment]
@janvennemann , 
 I followed the steps you mentioned in https://jira.appcelerator.org/browse/TIMOB-24446?focusedCommentId=419942&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-419942

But when I build with 6.2.0 SDK it fails to dexer:
```
[ERROR] :  Failed to run dexer:

[ERROR] :  

[ERROR] :  Uncaught translation error: java.lang.IllegalArgumentException: already added: Landroid/support/design/internal/package-info;

[ERROR] :  Uncaught translation error: java.lang.IllegalArgumentException: already added: Landroid/support/design/internal/BaselineLayout;

[ERROR] :  Uncaught translation error: java.lang.IllegalArgumentException: already added: Landroid/support/design/internal/BottomNavigationAnimationHelperBase;

[ERROR] :  Uncaught translation error: java.lang.IllegalArgumentException: already added: Landroid/support/design/internal/BottomNavigationAnimationHelperIcs;

[ERROR] :  Uncaught translation error: java.lang.IllegalArgumentException: already added: Landroid/support/design/internal/BottomNavigationItemView;

[ERROR] :  Uncaught translation error: java.lang.IllegalArgumentException: already added: Landroid/support/design/internal/BottomNavigationMenu;

[ERROR] :  Uncaught translation error: java.lang.IllegalArgumentException: already added: Landroid/support/design/internal/BottomNavigationMenuView;

[ERROR] :  Uncaught translation error: java.lang.IllegalArgumentException: already added: Landroid/support/design/internal/BottomNavigationMenuView$1;

[ERROR] :  Uncaught translation error: java.lang.IllegalArgumentException: already added: Landroid/support/design/internal/BottomNavigationPresenter;

[ERROR] :  

[ERROR] :  UNEXPECTED TOP-LEVEL EXCEPTION:

[ERROR] :  java.lang.RuntimeException: Translation has been interrupted

[ERROR] :  at com.android.dx.command.dexer.Main.processAllFiles(Main.java:615)

[ERROR] :  at com.android.dx.command.dexer.Main.runMultiDex(Main.java:368)

[ERROR] :  at com.android.dx.command.dexer.Main.runDx(Main.java:289)

[ERROR] :  at com.android.dx.command.dexer.Main.main(Main.java:247)

[ERROR] :  at com.android.dx.command.Main.main(Main.java:94)

[ERROR] :  Caused by: java.lang.InterruptedException: Too many errors

[ERROR] :  at com.android.dx.command.dexer.Main.processAllFiles(Main.java:607)

[ERROR] :  ... 4 more
 ```
Am I missing something here ?

**Comment by lokeshchdhry** [comment]
FR Passed.

Used the test steps here : https://jira.appcelerator.org/browse/TIMOB-24446?focusedCommentId=419942&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-419942

Studio Ver:     4.9.1.201707200100
SDK Ver:        6.2.0.v20170720154351 from  Jenkins in PR
OS Ver:         10.12.3
Xcode Ver:      Xcode 8.3.3
Appc NPM:       4.2.9
Appc CLI:       6.2.2
Ti CLI Ver:     5.0.14
Alloy Ver:      1.9.11
Node Ver:       6.10.1
Java Ver:       1.8.0_101
Devices:        Android 6.0, Android 7.0

**Pull request merged**